ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/mainnet/precompiles/BLAKE2fPrecompileContract.java

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.ethereum.mainnet.precompiles;
+
+import tech.pegasys.pantheon.ethereum.core.Gas;
+import tech.pegasys.pantheon.ethereum.mainnet.AbstractPrecompiledContract;
+import tech.pegasys.pantheon.ethereum.vm.GasCalculator;
+import tech.pegasys.pantheon.ethereum.vm.MessageFrame;
+import tech.pegasys.pantheon.util.bytes.BytesValue;
+
+import java.util.Arrays;
+
+import com.google.common.primitives.Ints;
+import com.google.common.primitives.Longs;
+import org.bouncycastle.util.Pack;
+
+public class BLAKE2fPrecompileContract extends AbstractPrecompiledContract {
```

Comment:
This name implies Compression Function F is on the same level as Blake2B and Blake2s.  `BLAKE2bCompressionFunctionFPrecompileContract` is a mouthful, but more accurate than `Blake2f`.  So I'de like to see it changed.  If solidity has a nice name for it we can use that one as `<foo>PrecompileContract`, so keep an eye out.